MapboxCoreNavigation/MGLMapboxEvents.swift

```diff
@@ -0,0 +1,39 @@
+import Mapbox
+import Polyline
+
+extension MGLMapboxEvents {
+    class func addDefaultEvents(routeProgress: RouteProgress, sessionIdentifier: UUID) -> [String: Any] {
+        var modifiedEventDictionary: [String: Any] = [:]
+    
+        modifiedEventDictionary["platform"] = UIDevice.current.systemName
+        modifiedEventDictionary["operatingSystemVersion"] = "\(UIDevice.current.systemName)-\(UIDevice.current.systemVersion)"
+        modifiedEventDictionary["sdkIdentifier"] = "mapbox-navigation-ios"
```

Comment:
It isn’t impossible: see the ideas in mapbox/MapboxDirections.swift#136, specifically `NSClassFromString(_:)` and `Bundle(for:)`.